### PR TITLE
E2E Test Framework Cleanup:

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**
- addresses several 'unchecked' warnings that cause yellow squigglies on Codium
- removes ignoreException() expressions that silence errors which we actually care about
- ensures that we wait for the taskDef to be properly created before we start the Task Worker.
- removes unneeded await's when running workflows in places where LH Server already guarantees RYOW

Overall, this should really reduce the time needed to complete the tests.

### CODE REVIEW

**Comment by coltmcnealy-lh** [comment]
Note that I also snuck in a Read-Your-Own-Writes test.

File: lh-test/src/main/java/io/littlehorse/test/LHExtension.java [hunk 1]
```diff
@@ -46,16 +47,19 @@ public void postProcessTestInstance(Object testInstance, ExtensionContext contex
...
-                        .ignoreException(LHMisconfigurationException.class)
-                        .until(() -> {
-                            try {
-                                worker.start();
-                                return true;
-                            } catch (IOException e) {
-                                throw new IllegalStateException(e);
-                            }
-                        });
+                        .ignoreExceptionsMatching(exn -> LHTestExceptionUtil.isNotFoundException(exn))
```
Comment (eduwercamacaro) [inline lh-test/src/main/java/io/littlehorse/test/LHExtension.java:53]:
```suggestion
                        .ignoreExceptionsMatching(LHTestExceptionUtil::isNotFoundException)
```

File: lh-test/src/main/java/io/littlehorse/test/LHExtension.java [hunk 2]
```diff
@@ -46,16 +47,19 @@ public void postProcessTestInstance(Object testInstance, ExtensionContext contex
...
+                        .until(() -> testContext.getLhClient().getTaskDef(taskDefId), taskDef -> taskDef != null);
```
Comment (eduwercamacaro) [inline lh-test/src/main/java/io/littlehorse/test/LHExtension.java:54]:
```suggestion
                        .until(() -> testContext.getLhClient().getTaskDef(taskDefId), Objects::nonNull);
```

File: lh-test/src/main/java/io/littlehorse/test/WfRunVerifier.java [hunk 3]
```diff
@@ -66,6 +67,10 @@ public WfRunVerifier thenVerifyTaskRunResult(
         return this;
     }
 
+    public WfRunVerifier thenVerifyAllTaskRuns(int threadRunNumber, Consumer<List<TaskRun>> verifier) {
```
Comment (eduwercamacaro) [inline lh-test/src/main/java/io/littlehorse/test/WfRunVerifier.java:70]: missing implementation?

File: lh-test/src/main/java/io/littlehorse/test/internal/TestContext.java [hunk 4]
```diff
@@ -146,19 +153,34 @@ public void registerUserTaskSchemas(Object testInstance) throws IllegalAccessExc
             }
             userTaskSchemasStore.put(taskDefRequest.getName(), userTaskSchema);
             registerUserTaskDef(taskDefRequest);
+            Awaitility.await()
+                    .ignoreExceptionsMatching(exn -> LHTestExceptionUtil.isNotFoundException(exn))
```
Comment (eduwercamacaro) [inline lh-test/src/main/java/io/littlehorse/test/internal/TestContext.java:157]:
```suggestion
                    .ignoreExceptionsMatching(LHTestExceptionUtil::isNotFoundException)
```

**Review by eduwercamacaro — APPROVED** [review]

File: server/src/test/java/e2e/ConditionalsTest.java [hunk 5]
```diff
@@ -318,6 +321,57 @@ private static Stream<Arguments> provideSuccessArguments() {
...
+                    .prepareRun(workflowNotIn, Arg.of("input", inputObject))
+                    .waitForStatus(LHStatus.COMPLETED)
+                    .thenVerifyTaskRunResult(0, 1, variableValue -> assertThat(variableValue.getBool())
+                            .isEqualTo(true))
+                    .thenVerifyTaskRunResult(0, 3, variableValue -> assertThat(variableValue.getBool())
+                            .isEqualTo(expectedOutput))
+                    .start();
+        }
+
+        private static Stream<Arguments> provideSuccessArguments() {
```
Comment (eduwercamacaro) [inline server/src/test/java/e2e/ConditionalsTest.java:341]: This wont work. verify test method arguments, it is expecting an instance of `InputObj` class

**Review by eduwercamacaro — CHANGES_REQUESTED** [review]
e2e tests are broken

File: lh-test/src/main/java/io/littlehorse/test/LHExtension.java (hunk 1 above)
Comment (coltmcnealy-lh) [reply lh-test/src/main/java/io/littlehorse/test/LHExtension.java:53]: For some reason, that didn't compile. I got a red underline on my editor.

File: lh-test/src/main/java/io/littlehorse/test/LHExtension.java (hunk 2 above)
Comment (coltmcnealy-lh) [reply lh-test/src/main/java/io/littlehorse/test/LHExtension.java:54]: Same here, I got a red underline...let me restart my editor

**Comment by coltmcnealy-lh** [comment]
Hang on I thought this was "draft" oops

**Comment by coltmcnealy-lh** [comment]
Ok now it's not a draft anymore @eduwercamacaro . This is ready to go. The `VerifyTaskRunOutputs` step is a little bit hacky. I will make another PR this week to move some more tests over to the new framework; when I do that I will improve `VerifyTaskRunOutputs` as needed.

**Review by eduwercamacaro — APPROVED** [review]

**Pull request merged**